msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/SynopticMsgcntrManagerImpl.java

```diff
@@ -70,48 +80,68 @@ public class SynopticMsgcntrManagerImpl extends HibernateDaoSupport implements S
 	private HashMap mfPageInSiteMap, sitesMap;
 	// transient variable for when on home page of site
 	private transient DecoratedCompiledMessageStats siteContents;
+
 	private MessageForumsMessageManager messageManager;
 	private UIPermissionsManager uiPermissionsManager;
+	private PreferencesService preferencesService;
+
+	@Getter
 	private PrivateMessageManager pvtMessageManager;
+
 	private MessageForumsTypeManager typeManager;
 	private DiscussionForumManager forumManager;
-	
+
 	/* Kernel */
 	private SqlService sqlService;
 	private SecurityService securityService;
 	private SiteService siteService;
 
-	public void setSecurityService(SecurityService securityService) {
-		this.securityService = securityService;
-	}
-
-	public void setSiteService(SiteService siteService) {
-		this.siteService = siteService;
-	}
-
-	public void setSqlService(SqlService sqlService) {
-		this.sqlService = sqlService;
-	}
-	
 	private static int ORACLE_IN_CLAUSE_SIZE_LIMIT = 1000;
 
-	
 	public SynopticMsgcntrManagerImpl() {}
-	
+
 	public void init() {
 		log.info("init()");
 	}
 
-	public List<SynopticMsgcntrItem> getWorkspaceSynopticMsgcntrItems(final String userId) {
+    public List<SynopticMsgcntrItem> getWorkspaceSynopticMsgcntrItems(final String userId) {
 
-		HibernateCallback<List<SynopticMsgcntrItem>> hcb = session -> {
+        HibernateCallback<List<SynopticMsgcntrItem>> hcb = session -> {
             Query q = session.getNamedQuery(QUERY_WORKSPACE_SYNOPTIC_ITEMS);
             q.setParameter("userId", userId, StringType.INSTANCE);
             return q.list();
         };
 
-		return getHibernateTemplate().execute(hcb);
-	}
+        List<SynopticMsgcntrItem> items = getHibernateTemplate().execute(hcb);
+
+        Preferences prefs = preferencesService.getPreferences(userId);
+        if (prefs != null) {
+            ResourceProperties props = prefs.getProperties(PreferencesService.SITENAV_PREFS_KEY);
+            if (props != null) {
+                int tabLabelPref = NumberUtils.toInt(props.getProperty(PreferencesService.TAB_LABEL_PREF), -1);
+                if (tabLabelPref == 2) {
```

Comment:
`2`? Do we have a constant for that so the expressions could be a bit more meaningful an maintainable?

Reply:
No we don't. I didn't bother as these are used so little. I might even add a method to the preferences service.

Reply:
Cool!